### PR TITLE
feat: Relocate computeSystemFee and refactor getModifiedFlows

### PR DESCRIPTION
**Pull request by nicholaspai**
- `computeLpFee` shouldn't try to retrieve a bundle config from state since its expected to be used by the relayer. If the relayer is trying to set the LP fee for a very recent deposit, then it should not be using the config included at the time of the last validated bundle, instead it should be using the latest config in the `ConfigStore` contract
- `getModifiedFlows` is refactored and now is used in `resolveHistoricalDepositForFill` to reduce the DOS vector
- The only remaining DOS vector is looking up an old fill for a refund. I've left a TODO

Fixes #ACX1321

### CODE REVIEW

File: src/clients/UBAClient/UBAClientUtilities.ts [hunk 1]
```diff
@@ -835,39 +834,43 @@ export type SpokePoolFillFilter = SpokePoolEventFilter & {
...
-    if (
-      (isDefined(originChainId) && fill.originChainId !== originChainId) ||
-      (isDefined(repaymentChainId) && fill.repaymentChainId !== repaymentChainId) ||
-      (isDefined(relayer) && fill.relayer !== relayer) ||
-      (isDefined(isSlowRelay) && fill.updatableRelayData.isSlowRelay !== isSlowRelay)
-    ) {
-      return false;
-    }
+  const fills = (
+    await mapAsync(spokePoolClient.getFills(), async (fill) => {
```
Comment (pxrl) [inline src/clients/UBAClient/UBAClientUtilities.ts:847]: What's the reason for switching to use `mapAsync` here insteead of `filterAsync` ? As far as I can see, `filterAsync` does what we want and doesn't require the subsequent loop to filter out all the `undefined` instances.

Comment (pxrl) [reply src/clients/UBAClient/UBAClientUtilities.ts:847]: Ah - I see, it's because `quoteBlockNumber`  is being added to each fill.

Comment (nicholaspai) [reply src/clients/UBAClient/UBAClientUtilities.ts:923]: yep

File: src/clients/UBAClient/UBAClientUtilities.ts [hunk 2]
```diff
@@ -297,6 +293,27 @@ export function getMostRecentBundleBlockRanges(
...
+export async function resolveCorrespondingDepositForFill(
+  fill: FillWithBlock,
+  spokePoolClients: SpokePoolClients,
+  hubPoolClient: HubPoolClient
+): Promise<DepositWithBlock | undefined> {
+  // Matched deposit for fill was not found in spoke client. This situation should be rare so let's
+  // send some extra RPC requests to blocks older than the spoke client's initial event search config
+  // to find the deposit if it exists.
+  return queryHistoricalDepositForFill(hubPoolClient, spokePoolClients, fill);
+}
```
Comment (james-a-morris) [inline src/clients/UBAClient/UBAClientUtilities.ts:315]: Is this currently just a wrapper over `resolveCorrespondingDepositForFill`

File: src/clients/UBAClient/UBAClientUtilities.ts [hunk 3]
```diff
@@ -672,24 +748,17 @@ async function getFlows(
...
+      (
+        await getFills(chainId, hubPoolClient, spokePoolClients, {
+          fromBlock,
+          toBlock,
+          repaymentChainId: chainId,
+          isSlowRelay: false,
+        })
+      ).filter((fill: FillWithBlock) => {
+        // We only want to include full fills as flows. Partial fills need to request refunds and those refunds
+        // will be included as flows.
```
Comment (james-a-morris) [inline src/clients/UBAClient/UBAClientUtilities.ts:760]: Could we use `filterAsync`

**Review by james-a-morris — COMMENTED** [review]
Left comments

File: src/clients/UBAClient/UBAClientUtilities.ts (hunk 2 above)
Comment (nicholaspai) [reply src/clients/UBAClient/UBAClientUtilities.ts:315]: yeah basically

File: src/clients/UBAClient/UBAClientUtilities.ts (hunk 3 above)
Comment (nicholaspai) [reply src/clients/UBAClient/UBAClientUtilities.ts:760]: this is a `sync` filter actually. The `array` is `await getFills` and the filter function is syncrhonous

**Review by james-a-morris — APPROVED** [review]
LGTM - pending @pxrl signoff and the resolution of my outstanding comment

File: src/clients/UBAClient/UBAClientUtilities.ts [hunk 4]
```diff
@@ -951,48 +923,36 @@ export async function getFills(
...
-      (isDefined(destinationChainId) && refundRequest.destinationChainId !== destinationChainId) ||
-      (isDefined(relayer) && refundRequest.relayer !== relayer)
-    ) {
-      return false;
-    }
-
-    const result = await refundRequestIsValid(spokePoolClients, hubPoolClient, refundRequest);
-
-    return result.valid;
-  });
```
Comment (pxrl) [inline src/clients/UBAClient/UBAClientUtilities.ts]: Can we retain this filtering? It's optional to use, and the relayer relies on filtering by `relayer` to find its existing refund requests.

Comment (nicholaspai) [reply src/clients/UBAClient/UBAClientUtilities.ts]: sure

**Pull request merged**